multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/ReferencePattern.java

```diff
@@ -26,6 +28,18 @@ protected boolean hasDepthOfReference() {
         return PLACEHOLDER.equals(this);
     }
 
+    private String getPattern() {
+        return pattern;
+    }
+
+    private static List<Pattern> getCompiledPatterns() {
+        List<Pattern> compiledPatterns = new ArrayList<>();
+        for (ReferencePattern referencePattern : ReferencePattern.values()) {
+            compiledPatterns.add(Pattern.compile(referencePattern.getPattern()));
```

Comment:
this is ok, but I though we could just make a change to the ReferencePattern enum constructor, to just compile the strings in place and then make the compiled patterns available, instaed of keep recompiling them even there
https://stackoverflow.com/questions/1720191/java-util-regex-importance-of-pattern-compile

Reply:
actually, we might not even want to keep the patterns here, since the enum can just list it's values, as long as the patterns are changed to be kept precompiled there just for optimization sake

Reply:
Added compiled patterns in the constructor, with the latest commit